test/extra/extraRewardsMultiMerkle.test.ts

```diff
@@ -0,0 +1,1137 @@
+
+            const claim_params = [
+                { 
+                    token: token1.address,
+                    index: 0,
+                    amount: claim_amounts1[0],
+                    merkleProof: tree.getProof(0, user1.address, claim_amounts1[0])
+                },
+                { 
+                    token: token2.address,
+                    index: 0,
+                    amount: claim_amounts2[0],
+                    merkleProof: tree2.getProof(0, user1.address, claim_amounts2[0])
+                }
+            ]
+
+            const token_nonce1 = await distributor.nonce(token1.address)
+            const token_nonce2 = await distributor.nonce(token2.address)
+
+            const old_balance1 = await token1.balanceOf(user1.address)
+            const old_balance2 = await token2.balanceOf(user1.address)
+
+            const claim_tx = await distributor.connect(user1).multiClaim(
+                user1.address,
+                claim_params
+            )
+    
+            await expect(
+                claim_tx
+            ).to.emit(distributor, "Claimed")
+                .withArgs(token1.address, 0, user1.address, claim_amounts1[0], token_nonce1);
+    
+            await expect(
+                claim_tx
+            ).to.emit(distributor, "Claimed")
+                .withArgs(token2.address, 0, user1.address, claim_amounts2[0], token_nonce2);
+
+            const new_balance1 = await token1.balanceOf(user1.address)
+            const new_balance2 = await token2.balanceOf(user1.address)
+
+            expect(new_balance1.sub(old_balance1)).to.be.eq(claim_amounts1[0])
+            expect(new_balance2.sub(old_balance2)).to.be.eq(claim_amounts2[0])
+
+            expect(await distributor.isClaimed(token1.address, 0)).to.be.true
+            expect(await distributor.isClaimed(token2.address, 0)).to.be.true
+
+
+        });
+
+        it(' should fail if 1 already claimed', async () => {
+
+            await distributor.connect(user1).claim(token1.address, 0, user1.address, claim_amounts1[0], tree.getProof(0, user1.address, claim_amounts1[0]))
+
+            const claim_params = [
+                { 
+                    token: token1.address,
+                    index: 0,
+                    amount: claim_amounts1[0],
+                    merkleProof: tree.getProof(0, user1.address, claim_amounts1[0])
+                },
+                { 
+                    token: token2.address,
+                    index: 0,
+                    amount: claim_amounts2[0],
+                    merkleProof: tree2.getProof(0, user1.address, claim_amounts2[0])
+                }
+            ]
+
+            await expect(
+                distributor.connect(user1).multiClaim(
+                    user1.address,
+                    claim_params
+                )
+            ).to.be.revertedWith('AlreadyClaimed')
+
+        });
+
+        it(' should fail if the claims are frozen', async () => {
```

Comment:
what happens if 1 token is frozen, but not the others

Reply:
should only block the claims on the frozen ones (all of the token do not need to be updated every time)

Reply:
you sure ? 
cuz
`if(frozen[token]) revert Errors.MerkleRootFrozen();`

Reply:
mb, misread method name.
In a `multiClaim`, if one of the token is frozen, the whole tx will indeed revert, since it's a loop of call to the `claim()` method, which reverts if the given token is frozen.